### PR TITLE
feat: compatibility

### PR DESCRIPTION
**Pull request by kellyjosephprice**
| [![PR App][icn]][demo] | Fix RM-9827 |
| :--------------------: | :---------: |

## 🧰 Changes

Adds backwards compatibility with variables, glossary, and reusable content.

When we're trying to update a document from markdown to MDX, we first parse the document to an mdast using the old markdown library. Then we serialize it back to MDX with this version. Basically:

```
const mdx = rmdx.mdx(rdmd.mdast(doc))
```

But, because these types are now completely represented in JSX, we didn't have any custom compilers to handle them. This adds some compilers to write their nodes back out as JSX.

### Callouts

Also, refactors how callouts are represented. Adds a prop to the node `empty` to represent if there's a heading a not. It's a bit more plumbing to get mdx to parse and execute props, so this leaves the heading as a child.

### HTML Comments

To be able to migrate away, we need to replace html comments with JSX comments.

## 🧬 QA & Testing

- [Broken on production][prod].
- [Working in this PR app][demo].

[demo]: https://markdown-pr-PR_NUMBER.herokuapp.com
[prod]: https://SUBDOMAIN.readme.io
[icn]: https://user-images.githubusercontent.com/886627/160426047-1bee9488-305a-4145-bb2b-09d8b757d38a.svg

### CODE REVIEW

**Review by trishaprile — APPROVED** [review]
goated

File: processor/compile/compatability.ts [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+    case NodeTypes.reusableContent:
+      return `<${node.tag} />`;
+    case 'html':
+      return node.value.replaceAll(/<!--(.*)-->/g, '{/*$1*/}');
+    default:
+      throw new Error('Unhandled node type!');
+  }
+};
+
+export default compatability;
```
Comment (jennspencer) [inline processor/compile/compatability.ts:25]: 😬 😬 😬 how do i break it to you that you spelled compatibility wrong 😬

Comment (kellyjosephprice) [reply processor/compile/compatability.ts:25]: If spelling compatability like compat-ability is wrong, I don't want to be right.

**Comment by rafegoldberg** [comment]
## This PR was released!

🚀 Changes included in [<b>v6.75.0-beta.38</b>](/readmeio/markdown/releases/tag/v6.75.0-beta.38)
<br>

**Pull request merged**